Goal: Be more verbose and indicate that the password was not changed when
      pam_chauthtok fails (in addition to the PAM error, which may not be
      comprehensible for the users)

Fixes: #352137

Status wrt upstream: not forwarded yet

Index: shadow-4.0.16/libmisc/pam_pass.c
===================================================================
--- shadow-4.0.16.orig/libmisc/pam_pass.c	2006-05-31 23:23:47.848384248 +0700
+++ shadow-4.0.16/libmisc/pam_pass.c	2006-05-31 23:25:11.939600440 +0700
@@ -38,6 +38,7 @@
 	ret = pam_chauthtok (pamh, flags);
 	if (ret != PAM_SUCCESS) {
 		fprintf (stderr, _("passwd: %s\n"), pam_strerror (pamh, ret));
+		fprintf (stderr, _("passwd: password unchanged\n"));
 		pam_end (pamh, ret);
 		exit (10);	/* XXX */
 	}
